File: spring-web/src/main/java/com/example/springwebex/service/AopExamService.java
package com.example.springwebex.service;

import org.springframework.stereotype.Component;
import com.example.springwebex.aop.AopExamAspectJointPoint;


@Component
public class AopExamService {
    public void methodNorm() {
        try {
            Thread.sleep(1000);
            System.out.println("joint point method - methodNorm");
        } catch (Exception e) {
            e.printStackTrace();
        }
    }

    @AopExamAspectJointPoint
    public void methodAspectDuplicated() {
        try {
            Thread.sleep(1000);
            System.out.println("joint point method - methodAspectDuplicated");
        } catch (Exception e) {
            e.printStackTrace();
        }
    }

    public void methodExcept() throws Exception {
        Thread.sleep(1000);
        System.out.println("joint point method - methodExcept");
        throw new Exception("This is Error...");
    }

}

File: spring-web-kotlin/src/main/kotlin/com/example/springwebex/service/AopExamService.kt
package com.example.springwebex.service

import org.springframework.stereotype.Component
import com.example.springwebex.aop.AopExam

@Component
class AopExamService {

    fun methodNorm() {
        try {
            Thread.sleep(1000)
            println("joint point method - methodNorm")
        } catch (e: Exception) {
            e.printStackTrace()
        }
    }

    @AopExam
    fun methodAspectDuplicated() {
        try {
            Thread.sleep(1000)
            println("joint point method - methodAspectDuplicated")
        } catch (e: Exception) {
            e.printStackTrace()
        }
    }

    fun methodExcept() {
        try {
            Thread.sleep(1000)
            println("joint point method - methodExcept")
            throw Exception("This is Error...")
        } catch (e: Exception) {
            throw e
        }
    }
}
